invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentAccessRestrictions/state/actions.js

```diff
@@ -0,0 +1,46 @@
+import {
+  SET_RESTRICTIONS_IS_LOADING,
+  SET_RESTRICTIONS_SUCCESS,
+  SET_RESTRICTIONS_HAS_ERROR,
+} from './types';
+import { document as documentApi } from '../../../../../../common/api';
+import {
+  sendErrorNotification,
+  sendSuccessNotification,
+} from '../../../../../../common/components/Notifications';
+import { fetchDocumentDetails } from '../../../state/actions';
+import { ES_DELAY } from '../../../../../../common/config';
+import isEmpty from 'lodash/isEmpty';
+
+export const setRestrictionsOnDocument = (pid, accessList) => {
+  const operation = isEmpty(accessList) ? 'remove' : 'add';
```

Comment:
Should there also be a `replace` operation if you update an existing list?

Reply:
Currently, if we do an "add" to "/_access/read" and submit a list, it will overwrite whatever is there (if any)...and the existing access list is always pre-populated.

I don't think we need it